forge/forge/compile.py

```diff
@@ -267,6 +267,9 @@ def forge_compile_from_context(context: CompileContext) -> CompiledModel:
         CompileDepth.FINISH_COMPILE: finish_compile,
     }
 
+    if context.loss_module is not None:
```

Comment:
How does this check work if the loss module is on the CPU? I also see that there is a specific `training` flag. Can we use it, or something similar?

Reply:
We require and expect loss and optimizer modules to be added as `forge.compile()` arguments, so that's not the issue. Anyhow, it's better to check the `training` flag, so I've modified the code.